pkg/commands/compute/language_javascript.go

```diff
@@ -0,0 +1,190 @@
+			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s", text.Bold("npm init")),
+		}
+	}
+
+	fmt.Fprintf(out, "Found package.json at %s\n", fpath)
+	fmt.Fprintf(out, "Installing package dependencies...\n")
+
+	cmd := fstexec.Streaming{
+		Command: "npm",
+		Args:    []string{"install"},
+		Env:     []string{},
+		Output:  out,
+	}
+	return cmd.Exec()
+}
+
+// Verify implements the Toolchain interface and verifies whether the
+// JavaScript language toolchain is correctly configured on the host.
+func (a JavaScript) Verify(out io.Writer) error {
+	// 1) Check `npm` is on $PATH
+	//
+	// npm is Node/JavaScript's toolchain installer and manager, it is
+	// needed to assert that the correct versions of the ecp-js compiler and
+	// @fastly/js-compute package are installed. We only check whether the
+	// binary exists on the users $PATH and error with installation help text.
+	fmt.Fprintf(out, "Checking if npm is installed...\n")
+
+	p, err := exec.LookPath("npm")
+	if err != nil {
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("`npm` not found in $PATH"),
+			Remediation: fmt.Sprintf("To fix this error, install Node.js and npm by visiting:\n\n\t$ %s", text.Bold("https://nodejs.org/")),
+		}
+	}
+
+	fmt.Fprintf(out, "Found npm at %s\n", p)
+
+	// 2) Check package.json file exists in $PWD
+	//
+	// A valid npm package is needed for compilation and to assert whether the
+	// required dependencies are installed locally. Therefore, we first assert
+	// whether one exists in the current $PWD.
+	fpath, err := filepath.Abs("package.json")
+	if err != nil {
+		return fmt.Errorf("getting package.json path: %w", err)
+	}
+
+	if !filesystem.FileExists(fpath) {
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("package.json not found"),
+			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s", text.Bold("npm init")),
+		}
+	}
+
+	fmt.Fprintf(out, "Found package.json at %s\n", fpath)
+
+	// 3) Check if `ecp-js` is installed.
+	//
+	// ecp-js is the JavaScript compiler. We first check if the required
+	// dependency exists in the package.json and then whether the ecp-js binary
+	// exists in the npm bin directory.
+	fmt.Fprintf(out, "Checking if @fastly/js-compute is installed...\n")
+	if !checkPackageDependencyExists("@fastly/js-compute") {
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("`@fastly/js-compute` not found in package.json"),
+			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s", text.Bold("npm install --save-dev @fastly/js-compute")),
+		}
+	}
+
+	p, err = getNpmBinPath()
+	if err != nil {
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("could not determine npm bin path"),
+			Remediation: fmt.Sprintf("To fix this error, run the following command:\n\n\t$ %s", text.Bold("npm install --global npm@latest")),
+		}
+	}
+
+	path, err := exec.LookPath(filepath.Join(p, "ecp-js"))
```

Comment:
`ecp-js` was renamed to `js-compute-runtime`, so this and the other mentions should be updated accordingly